data/src/main/java/com/linkedin/data/schema/AbstractSchemaParser.java

```diff
@@ -0,0 +1,252 @@
+/*
+   Copyright (c) 2012 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.data.schema;
+
+
+import com.linkedin.data.codec.DataLocation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Common base class for schema parsers.
+ *
+ * @author slim
+ * @author jbetz
+ */
+abstract public class AbstractSchemaParser extends AbstractDataParser implements PegasusSchemaParser
```

Comment:
Thanks for update. I will download the diff and run some sanity check before committing to master, will let you know if I run into any issues.